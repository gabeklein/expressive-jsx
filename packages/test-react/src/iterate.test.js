const test = require("./_adapter");

test("for statement", `
  () => {
    for(let i = 0; i < 10; i++)
      <div>Number: {i}</div>
  };
`);

test("for-in statement", `
  () => {
    for(const item in group)
      <div>{group[item]}</div>
  };
`);

test("for-of statement", `
  () => {
    for(const item of group)
      <div>{item}</div>
  };
`);

test("for-of statement; key defined", `
  () => {
    for(const item of group)
      <div key={item}>{item}</div>
  };
`);