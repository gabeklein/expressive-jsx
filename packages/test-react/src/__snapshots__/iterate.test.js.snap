// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`for statement 1`] = `
"() => collect(add => {
  for (let i = 0; i < 10; i++) add(create(\\"div\\", {}, \\"Number: \\", i));
});"
`;

exports[`for statement 2`] = `
"() => collect(add => {
  for (let i = 0; i < 10; i++) add(<div>Number: {i}</div>);
});"
`;

exports[`for-in statement 1`] = `
"() => Object.keys(group).map(item => create(\\"div\\", {
  key: item
}, group[item]));"
`;

exports[`for-in statement 2`] = `"() => Object.keys(group).map(item => <div key={item}>{group[item]}</div>);"`;

exports[`for-of statement 1`] = `
"() => group.map((item, i) => create(\\"div\\", {
  key: i
}, item));"
`;

exports[`for-of statement 2`] = `"() => group.map((item, i) => <div key={i}>{item}</div>);"`;

exports[`for-of statement; key defined 1`] = `
"() => group.map(item => create(\\"div\\", {
  key: item
}, item));"
`;

exports[`for-of statement; key defined 2`] = `"() => group.map(item => <div key={item}>{item}</div>);"`;
